From 152217064f1192bb836620ad7eaadbc031624f4c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=ADctor=20Manuel=20J=C3=A1quez=20Leal?=
 <vjaquez@igalia.com>
Date: Mon, 20 Mar 2017 16:45:01 +0100
Subject: [PATCH] plugins: when debug disabled, default category is NULL

As in gstreamer-vaapi a common base class is used, the specific
default category is passed to the base-plugin initializator, thus
the log messages are categorized with the used plugin.

Nonetheless, when the gst-debug is disabled in compilation time,
it is needed to pass NULL to the base-plugin initializator. This
patch does that.

https://bugzilla.gnome.org/show_bug.cgi?id=780302

Signed-off-by: Rahul Bedarkar <rahul.bedarkar@imgtec.com>
---
 gst/vaapi/gstvaapidecode.c   | 4 ++++
 gst/vaapi/gstvaapiencode.c   | 4 ++++
 gst/vaapi/gstvaapipostproc.c | 4 ++++
 gst/vaapi/gstvaapisink.c     | 4 ++++
 4 files changed, 16 insertions(+)

diff --git a/gst/vaapi/gstvaapidecode.c b/gst/vaapi/gstvaapidecode.c
index 407e25b..c2bb42a 100644
--- a/gst/vaapi/gstvaapidecode.c
+++ b/gst/vaapi/gstvaapidecode.c
@@ -49,7 +49,11 @@
 #define GST_VAAPI_DECODE_FLOW_PARSE_DATA        GST_FLOW_CUSTOM_SUCCESS_2
 
 GST_DEBUG_CATEGORY_STATIC (gst_debug_vaapidecode);
+#ifndef GST_DISABLE_GST_DEBUG
 #define GST_CAT_DEFAULT gst_debug_vaapidecode
+#else
+#define GST_CAT_DEFAULT NULL
+#endif
 
 #define GST_VAAPI_DECODE_PARAMS_QDATA \
   g_quark_from_static_string("vaapidec-params")
diff --git a/gst/vaapi/gstvaapiencode.c b/gst/vaapi/gstvaapiencode.c
index dd60901..b0f0a32 100644
--- a/gst/vaapi/gstvaapiencode.c
+++ b/gst/vaapi/gstvaapiencode.c
@@ -38,7 +38,11 @@
 #define GST_VAAPI_ENCODE_FLOW_CONVERT_ERROR     GST_FLOW_CUSTOM_ERROR_1
 
 GST_DEBUG_CATEGORY_STATIC (gst_vaapiencode_debug);
+#ifndef GST_DISABLE_GST_DEBUG
 #define GST_CAT_DEFAULT gst_vaapiencode_debug
+#else
+#define GST_CAT_DEFAULT NULL
+#endif
 
 G_DEFINE_ABSTRACT_TYPE_WITH_CODE (GstVaapiEncode,
     gst_vaapiencode, GST_TYPE_VIDEO_ENCODER,
diff --git a/gst/vaapi/gstvaapipostproc.c b/gst/vaapi/gstvaapipostproc.c
index f4ddbb5..d45f7ee 100644
--- a/gst/vaapi/gstvaapipostproc.c
+++ b/gst/vaapi/gstvaapipostproc.c
@@ -49,7 +49,11 @@
 #define GST_PLUGIN_DESC "A VA-API video postprocessing filter"
 
 GST_DEBUG_CATEGORY_STATIC (gst_debug_vaapipostproc);
+#ifndef GST_DISABLE_GST_DEBUG
 #define GST_CAT_DEFAULT gst_debug_vaapipostproc
+#else
+#define GST_CAT_DEFAULT NULL
+#endif
 
 /* Default templates */
 /* *INDENT-OFF* */
diff --git a/gst/vaapi/gstvaapisink.c b/gst/vaapi/gstvaapisink.c
index cca71d9..9cc5154 100644
--- a/gst/vaapi/gstvaapisink.c
+++ b/gst/vaapi/gstvaapisink.c
@@ -59,7 +59,11 @@
 #define GST_PLUGIN_DESC "A VA-API based videosink"
 
 GST_DEBUG_CATEGORY_STATIC (gst_debug_vaapisink);
+#ifndef GST_DISABLE_GST_DEBUG
 #define GST_CAT_DEFAULT gst_debug_vaapisink
+#else
+#define GST_CAT_DEFAULT NULL
+#endif
 
 /* Default template */
 /* *INDENT-OFF* */
-- 
2.6.2

